services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreDeletionValidationUtils.java

```diff
@@ -0,0 +1,331 @@
+          .collect(Collectors.toList());
+
+      for (VeniceSystemStoreType systemStoreType: systemStoreTypesToCheck) {
+        final String systemStoreName = systemStoreType.getSystemStoreName(storeName);
+        final Store systemStore = admin.getStore(clusterName, systemStoreName);
+        if (systemStore != null) {
+          result.setStoreNotDeleted(
+              String.format("System store metadata still exists in storeRepository: %s", systemStoreName));
+          return true;
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Failed to check system store metadata for store: {} in cluster: {}", storeName, clusterName, e);
+      result.setStoreNotDeleted("Failed to check system store metadata: " + e.getMessage());
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Checks if Kafka topics related to the store still exist.
+   */
+  private static boolean checkForAnyExistingTopicResources(
+      Admin admin,
+      StoreDeletedValidation result,
+      String clusterName,
+      String storeName) {
+    try {
+      final TopicManager topicManager = admin.getTopicManager();
+      final Set<PubSubTopic> kafkaTopics = topicManager.listTopics();
+
+      final List<VeniceSystemStoreType> systemStoreTypesToCheck = Arrays.asList(VeniceSystemStoreType.values())
+          .stream()
+          .filter(type -> type != VeniceSystemStoreType.BATCH_JOB_HEARTBEAT_STORE)
+          .collect(Collectors.toList());
+
+      for (PubSubTopic topic: kafkaTopics) {
+        final String topicName = topic.getName();
+        if (isStoreRelatedTopic(topicName, storeName, systemStoreTypesToCheck)) {
+          result.setStoreNotDeleted(String.format("Kafka topic still exists: %s", topicName));
+          return true;
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Failed to check Kafka topics for store: {} in cluster: {}", storeName, clusterName, e);
+      result.setStoreNotDeleted("Failed to check Kafka topics: " + e.getMessage());
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Checks if Helix resources for the main store still exist.
+   */
+  private static boolean checkMainStoreHelixResources(
+      Admin admin,
+      StoreDeletedValidation result,
+      String clusterName,
+      String storeName) {
+    try {
+      // Since we can't use isResourceStillAlive directly with store name,
+      // and containsHelixResource isn't available in Admin interface,
+      // we'll check if any Kafka topics exist for this store (which would indicate Helix resources)
+      final TopicManager topicManager = admin.getTopicManager();
+      final Set<PubSubTopic> kafkaTopics = topicManager.listTopics();
+
+      for (PubSubTopic topic: kafkaTopics) {
+        final String topicName = topic.getName();
+        // Check for version topics that belong to this store
+        if (topicName.startsWith(storeName + "_v")) {
```

Comment:
Replace with `PubSubTopic::isVersionTopic`

Reply:
I strongly recommend checking the Venice Javadoc before adding custom logic. There may already be APIs available to achieve what you need. e.g. https://venicedb.org/javadoc/com/linkedin/venice/pubsub/api/PubSubTopic.html